internal/compiler/filesparser.go

```diff
@@ -138,132 +130,262 @@ func (t *parseTask) loadAutomaticTypeDirectives(loader *fileLoader) {
+	}
 
-func (w *filesParser) collectWorker(loader *fileLoader, tasks []*parseTask, iterate func(*parseTask), seen collections.Set[*parseTask]) {
-	for _, task := range tasks {
-		// Exclude automatic type directive tasks from include reason processing,
-		// as these are internal implementation details and should not contribute
-		// to the reasons for including files.
-		if task.redirectedParseTask == nil && !task.isForAutomaticTypeDirective {
+	loader.includeProcessor.fileIncludeReasons = make(map[tspath.Path][]*FileIncludeReason, totalFileCount)
+	var outputFileToProjectReferenceSource map[tspath.Path]string
+	if !loader.opts.canUseProjectReferenceSource() {
+		outputFileToProjectReferenceSource = make(map[tspath.Path]string, totalFileCount)
+	}
+	resolvedModules := make(map[tspath.Path]module.ModeAwareCache[*module.ResolvedModule], totalFileCount+1)
+	typeResolutionsInFile := make(map[tspath.Path]module.ModeAwareCache[*module.ResolvedTypeReferenceDirective], totalFileCount)
+	sourceFileMetaDatas := make(map[tspath.Path]ast.SourceFileMetaData, totalFileCount)
+	var jsxRuntimeImportSpecifiers map[tspath.Path]*jsxRuntimeImportSpecifier
+	var importHelpersImportSpecifiers map[tspath.Path]*ast.Node
+	var sourceFilesFoundSearchingNodeModules collections.Set[tspath.Path]
+	libFilesMap := make(map[tspath.Path]*LibFile, libFileCount)
+
+	var collectFiles func(tasks []*parseTask, seen map[*parseTaskData]string)
+	collectFiles = func(tasks []*parseTask, seen map[*parseTaskData]string) {
+		for _, task := range tasks {
 			includeReason := task.includeReason
-			if task.loadedTask != nil {
-				task = task.loadedTask
+			// Exclude automatic type directive tasks from include reason processing,
+			// as these are internal implementation details and should not contribute
+			// to the reasons for including files.
+			if task.redirectedParseTask == nil && !task.isForAutomaticTypeDirective {
+				if task.loadedTask != nil {
+					task = task.loadedTask
+				}
+				w.addIncludeReason(loader, task, includeReason)
+			}
+			data, _ := w.taskDataByPath.Load(task.path)
+			if !task.loaded {
+				continue
+			}
+
+			// ensure we only walk each task once
+			if checkedName, ok := seen[data]; ok {
+				if !loader.opts.Config.CompilerOptions().ForceConsistentCasingInFileNames.IsFalse() {
+					// Check if it differs only in drive letters its ok to ignore that error:
+					checkedAbsolutePath := tspath.GetNormalizedAbsolutePathWithoutRoot(checkedName, loader.comparePathsOptions.CurrentDirectory)
+					inputAbsolutePath := tspath.GetNormalizedAbsolutePathWithoutRoot(task.normalizedFilePath, loader.comparePathsOptions.CurrentDirectory)
+					if checkedAbsolutePath != inputAbsolutePath {
+						loader.includeProcessor.addProcessingDiagnosticsForFileCasing(task.path, checkedName, task.normalizedFilePath, includeReason)
+					}
+				}
+				continue
+			} else {
+				seen[data] = task.normalizedFilePath
+			}
+
+			if tasksSeenByNameIgnoreCase != nil {
```

Comment:
To make sure I understand this, why can't this check be made above along with the check for when `ForceConsistentCasingInFileNames` is true?

Reply:
Because in case sensitive systems, path= file name so they are distinct and need to check if lowecasing file names as path (as it would do on case insensitive file systems ) can have duplicates and report errors